src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -724,6 +724,15 @@ internal static bool WriteConsideredUse(TypeSymbol type, BoundExpression value)
                 return value.ConstantValue != ConstantValue.Null;
             }
 
+            if (type?.SpecialType == SpecialType.System_String && value.Kind == BoundKind.InterpolatedString)
```

Comment:
>if (type?.SpecialType == SpecialType.System_String && value.Kind == BoundKind.InterpolatedString) [](start = 12, length = 97)

Since this is a patch for a very specific scenario when an interpolated string has a constant value, I think the logic should identify the situation precisely and shouldn't return early, but rather adjust the ```if (value.ConstantValue != null) return false;``` to not return for an interpolated string. It also feels like it should be sufficient to simply check for ```value.Kind == BoundKind.InterpolatedString``` and not bother checking the type.